### PR TITLE
Hrcpp 440/dotnet core

### PR DESCRIPTION
**Pull request by rigazilla**
This enables the .NET Core build. Some fixes  and improvements has been also included and can't be easily organized in separated PR.

Issues:

https://issues.jboss.org/browse/HRCPP-440
https://issues.jboss.org/browse/HRCPP-441
https://issues.jboss.org/browse/HRCPP-442
https://issues.jboss.org/browse/HRCPP-443
https://issues.jboss.org/browse/HRCPP-444

### CODE REVIEW

File: buildtools/build.fsx [hunk 1]
```diff
@@ -67,45 +68,45 @@ Target "BuildSwigWraper" (fun _ ->
...
 Target "Generate" (fun _ ->
-    trace "proto files and swig files generated"
+    trace "Target Generate: proto files and swig files generated"
 )
 
 Target "Build" (fun _ ->
     Build (fun p -> { p with Project = "../Infinispan.HotRod.sln"
                              Configuration = "RelWithDebInfo"})
-    trace "solution built"
+    trace "Target Generate: solution built"
```
Comment (mgencur) [inline buildtools/build.fsx]: Should be Target Build?

File: README.md [hunk 2]
```diff
@@ -88,16 +88,19 @@ Requirements:
...
 
-#### Configuration
+#### Link the correct C++ core
 The C++ client library must match the compiler version in use for the swig wrapper. If the libraries download by the script (from ci.infinispan.org) do not fit your needs just compile the C++ client from the source and point the HOTROD_PREBUILT_DIR to the dist directory (the dir containing the unpacked .rpm), thus:
     HOTROD_PREBUILD_DIR=<path to unpacked rpm> ./build.sh
+    & { $env:HOTROD_PREBUILT_DIR="<path to unpacked zip> ; .\build.ps1 }
+
+### Run the test suite
+The system must be able to find the native libraries under the swig/native_client/lib folder so add it either to the LD_LIBRARY_PATH (linux) or to the PATH (win) variable. Define a JBOSS_HOME pointing to your Infinispan server location (in Windows use the Unix file separator '/') and then run the test suite with dotnet:
+	dotnet test
```
Comment (mgencur) [inline README.md]: dotnet test? I can't see a script or executable with name "dotnet".

**Comment by mgencur** [comment]
One more note: The WindowsTargetPlatformVersion tag in swig/hotrod_wrap.vcxproj has a hard-coded version: 10.0.14393.0.  I have to have this exact Microsoft SDK version installed in order to get it built otherwise I'm getting an error when building the swig wrapper. What about putting this among requirements in the readme or making it more flexible and allowing more versions? Not sure how this can be done though

File: README.md [hunk 3]
```diff
@@ -95,6 +95,9 @@ Once you run `.\build.ps1`, it will create a cache with all tools necessary, so
 
 To build make build.sh executable (chmod +x build.sh) and run `./build.sh Build` (any command mentioned here should work with `build.sh` instead of `build.ps1`).
 
+#### Configuration
+The C++ client library must match the compiler version in use for the swig wrapper. If the libraries download by the script (from ci.infinispan.org) do not fit your needs just compile the C++ client from the source and point the HOTROD_PREBUILT_DIR to the dist directory (the dir containing the unpacked .rpm), thus:
+    HOTROD_PREBUILD_DIR=<path to unpacked rpm> ./build.sh
```
Comment (mgencur) [inline README.md]: HOTROD_PREBUILD_DIR -> HOTROD_PREBUILT_DIR

**Comment by mgencur** [comment]
OK. I was able to build the client in both ways and run the tests in both ways as well. I think this is looking good except for the few items I mentioned above.

File: src/Infinispan.HotRod/Infinispan.HotRod.csproj [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
[the remark this answers is not in the document]
Comment (rigazilla) [reply src/Infinispan.HotRod/Infinispan.HotRod.csproj]: This one should specify also the .nuget package content. I general I prefer to keep the two build systems separated for now: I'm not sure I can handle a cmake+fake system :)

File: test/Infinispan.HotRod.Tests/AsyncOperationsTest.cs [hunk 5]
```diff
@@ -6,10 +6,11 @@
 using Infinispan.HotRod.Tests.Util;
 
 
-namespace Infinispan.HotRod.Tests.ClusteredSaslCsXml2
+namespace Infinispan.HotRod.Tests.StandaloneXml
 {
     [TestFixture]
-    [Category("clustered_sasl_cs_xml_2")]
+    [Category("standalone_xml")]
```
[the remark this answers is not in the document]
Comment (rigazilla) [reply test/Infinispan.HotRod.Tests/AsyncOperationsTest.cs]: Developer  can  use the standalone_xml category to run all the tests related to the standalone.xml configuration

**Comment by mgencur** [comment]
This PR now looks good. Integrating.

**Pull request merged**